rmw_zenoh_cpp/src/detail/graph_cache.hpp

```diff
@@ -36,43 +38,42 @@
 ///=============================================================================
 // TODO(Yadunund): Since we reuse pub_count_ and sub_count_ for pub/sub and
 // service/client consider more general names for these fields.
-struct TopicStats
-{
-  // The count of publishers or clients.
-  std::size_t pub_count_;
-
-  // The count of subscriptions or services.
-  std::size_t sub_count_;
-
-  // Constructor which initializes counters to 0.
-  TopicStats(std::size_t pub_count, std::size_t sub_count);
-};
-
-///=============================================================================
+// Consider changing this to an array of unordered_set where the index of the
+// array corresponds to the EntityType enum. This way we don't need to mix
+// pub/sub with client/service.
```

Comment:
Maybe change this to a `TODO` comment?  Because I also agree that refactoring it in this way would be clearer.

Reply:
https://github.com/ros2/rmw_zenoh/pull/103/commits/f768e32ef42e0c1e9c3ad7769f1c33b98c9344be